moveit_ros/planning/rdf_loader/src/rdf_loader.cpp

```diff
@@ -50,6 +50,11 @@
 #include <streambuf>
 #include <algorithm>
 
+#ifdef _WIN32
+#define popen _popen
+#define pclose _pclose
+#endif
+
```

Comment:
I agreed it is less optimal to use macro redefine for this case. Unfortunately, not every` POSIX` function names have the original names defined in MSVC C run-time library. In this case, using `popen` and `pclose` with MSVC actually yields compile errors, since there is no such function declarations. (opposing to just generate warnings.)

Reply:
this is somewhat frustrating indeed.
I guess we could provide a `windows_compatibility.h` somewhere to define such aliases, but that's probably too much overhead for now.

Reply:
Actually, I would argue to include such a compatibility header as part of this PR.

Reply:
I propose to have a compatibility header, providing inline function definitions for `popen` and `pclose`, which both forward to their underscore counterparts. That's a cleaner approach than using defines.

Reply:
The reason why I did not propose this is that these aliases depend on varying includes too
(`_chdir` apparently depends on `direct.h`) and I would not include such headers there prematurely.
But adding the `#define`'s without the functions being defined seems wrong too.

Reply:
My key point is to avoid defines. For the issue you raised, I see the following options:
You could write the compatibility header without including those other headers, but
1. rely on the developer to include them before the compat header when necessary
2. or explicitly declare those required functions in the compat header.